Framework/DataHandling/test/ISISJournalTest.h

```diff
@@ -0,0 +1,304 @@
+    auto journal = makeJournal();
+    auto results = journal.getRuns(valueNames(), multipleFilters());
+    auto const expected =
+        std::vector<ISISJournal::RunData>{{{"name", "INTER12346"},
+                                           {"run_number", "12346"},
+                                           {"title", "Experiment 1 run 2"}}};
+    TS_ASSERT_EQUALS(results, expected);
+    verifyAndClear();
+  }
+
+  void
+  test_getRuns_returns_requested_values_for_all_entries_when_no_filter_is_set() {
+    auto journal = makeJournal();
+    auto results = journal.getRuns(valueNames(), emptyFilters());
+    auto const expected =
+        std::vector<ISISJournal::RunData>{{{"name", "INTER22345"},
+                                           {"run_number", "22345"},
+                                           {"title", "Experiment 2 run 1"}},
+                                          {{"name", "INTER12345"},
+                                           {"run_number", "12345"},
+                                           {"title", "Experiment 1 run 1"}},
+                                          {{"name", "INTER12346"},
+                                           {"run_number", "12346"},
+                                           {"title", "Experiment 1 run 2"}}};
+    TS_ASSERT_EQUALS(results, expected);
+    verifyAndClear();
+  }
+
+  void test_getCycleList_with_empty_file_throws() {
+    auto journal = makeJournal(emptyFile);
+    TS_ASSERT_THROWS(journal.getCycleNames(), std::runtime_error const &);
+    verifyAndClear();
+  }
+
+  void test_getCycleList_with_bad_xml_throws() {
+    auto journal = makeJournal(badFile);
+    TS_ASSERT_THROWS(journal.getCycleNames(), std::runtime_error const &);
+    verifyAndClear();
+  }
+
+  void test_getCycleList_with_empty_xml_file_returns_empty_results() {
+    auto journal = makeJournal(emptyIndexFile);
+    auto results = journal.getCycleNames();
+    TS_ASSERT_EQUALS(results, std::vector<std::string>{});
+    verifyAndClear();
+  }
+
+  void test_getCycleList_throws_when_invalid_element_names() {
+    auto journal = makeJournal(invalidIndexFile);
+    TS_ASSERT_THROWS(journal.getCycleNames(), std::invalid_argument const &);
+    verifyAndClear();
+  }
+
+  void test_getCycleList_returns_all_valid_cycles() {
+    auto journal = makeJournal(indexFile);
+    auto results = journal.getCycleNames();
+    auto const expected =
+        std::vector<std::string>{"17_1", "18_1", "19_1", "19_2"};
+    TS_ASSERT_EQUALS(results, expected);
+    verifyAndClear();
+  }
+
+private:
+  NiceMock<MockInternetHelper> *m_internetHelper;
+
+  ISISJournal
+  makeJournal(std::string const &xmlContents = std::string(journalFile)) {
+    auto internetHelper =
+        std::make_unique<NiceMock<MockInternetHelper>>(xmlContents);
+    m_internetHelper = internetHelper.get();
+    auto status = Poco::Net::HTTPResponse::HTTP_OK;
+    ON_CALL(*m_internetHelper, sendRequestProxy(_, _))
```

Comment:
I got a little confused about the lifetime here since it looked liked `m_internetHelper` dropped out of scope
Could we change `*m_internetHelper` to `std::move(m_internetHelper)`

Reply:
Not sure what you mean - it is std::moved in the creation of the ISISJournal. I agree this is not very clear but I haven't found a better way to do it...